typing/types.ml

```diff
@@ -322,6 +322,15 @@ and constructor_tag =
   | Cstr_extension of Path.t * bool     (* Extension constructor
                                            true if a constant false if a block*)
 
+let equal_tag t1 t2 = 
+  match (t1, t2) with
+  | Cstr_constant i1, Cstr_constant i2 -> i2 = i1
+  | Cstr_block i1, Cstr_block i2 -> i2 = i1
+  | Cstr_unboxed, Cstr_unboxed -> true
+  | Cstr_extension (path1, b1), Cstr_extension (path2, b2) -> 
+      Path.same path1 path2 && b1 = b2
+  | (Cstr_constant _|Cstr_block _|Cstr_unboxed|Cstr_extension _), _ -> false
+
```

Comment:
actually I find this more readable, should compiles fast and generate efficient code
```ocaml
let equal_tag t1 =
   match t1 with 
   | Cstr_constant i1 -> 
      begin match t2 with Cstr_constant i2 -> i1 = i2 
       | _ -> false 
      end
  | ..
      
```